openupgrade_scripts/scripts/website_sale/14.0.1.0/post-migration.py

```diff
@@ -1,10 +1,36 @@
 # Copyright (C) 2021 Open Source Integrators <https://www.opensourceintegrators.com/>
 # Copyright 2021 ForgeFlow S.L.  <https://www.forgeflow.com>
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import re
 
 from openupgradelib import openupgrade
 
 
+def extract_custom_product_page_term_conditions(env):
+    """Replace Terms and Conditions content in the new v14 template so as not to lose
+    content from previous versions if it has been customised."""
+    product_custom_text_view = env.ref("website_sale.product_custom_text")
+    product_custom_text_arch = product_custom_text_view.arch_db
+    product_custom_text_pattern = r'<p class="text-muted(.*?)<\/p>'
+    product_custom_text_content, *_ = re.findall(
+        product_custom_text_pattern, product_custom_text_arch, re.DOTALL
+    )
+    product_views = env["ir.ui.view"].search(
+        [("key", "=", "website_sale.product"), ("website_id", "!=", False)]
+    )
+    for view in product_views:
+        product_arch = view.arch_db
+        product_pattern = r'<hr\s*/>\s*<p\s+class="text-muted">(.*?)</p>'
+        product_matches, *_ = re.findall(product_pattern, product_arch, re.DOTALL)
```

Comment:
You can't do it here. If there's no match you'll raise unpack error.

Reply:
thanks :)